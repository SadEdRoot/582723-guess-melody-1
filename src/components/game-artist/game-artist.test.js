import React from 'react';
import GameArtist from './game-artist.jsx';
import renderer from 'react-test-renderer';

const mocks = {
  "type": `artist`,
  "song": {
    "artist": `Quincas Moreira`,
    "src": `https://es31-server.appspot.com/guess-melody/static/music/Blue_Whale.mp3`
  },
  "answers": [
    {
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg`,
      "artist": `Quincas Moreira`
    },
    {
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg`,
      "artist": `Density & Time`
    },
    {
      "picture": `https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg`,
      "artist": `Endless Love`
    }
  ]
};

it(`GameArtist correctly renders`, () => {
  const tree = renderer.create(<GameArtist
    gameTime={0}
    errorCount={0}
    question={mocks}
    onSubmit={()=>{}}
  />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
