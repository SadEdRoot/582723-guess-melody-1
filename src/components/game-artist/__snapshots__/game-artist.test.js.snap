// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameArtist correctly renders 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="390"
        cy="390"
        filter="url(#blur)"
        r="370"
        style={
          Object {
            "MsTransformOrigin": "center",
            "WebkitTransformOrigin": "center",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        0
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    />
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <button
        className="track__button track__button--play"
        disabled={true}
        onClick={[Function]}
        type="button"
      />
      <div
        className="track__status"
      >
        <audio />
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[Function]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-0"
          name="answer"
          type="radio"
          value="artist-0"
        />
        <label
          className="artist__name"
          htmlFor="artist-0"
        >
          <img
            alt="Quincas Moreira"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Quincas_Moreira.jpg"
          />
          Quincas Moreira
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-1"
          name="answer"
          type="radio"
          value="artist-1"
        />
        <label
          className="artist__name"
          htmlFor="artist-1"
        >
          <img
            alt="Density & Time"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Density_n_Time.jpg"
          />
          Density & Time
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="artist-2"
          name="answer"
          type="radio"
          value="artist-2"
        />
        <label
          className="artist__name"
          htmlFor="artist-2"
        >
          <img
            alt="Endless Love"
            className="artist__picture"
            src="https://es31-server.appspot.com/guess-melody/static/artist/Endless_Love.jpg"
          />
          Endless Love
        </label>
      </div>
    </form>
  </section>
</section>
`;
